sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1545,14 +1558,14 @@ case class TruncTimestamp(
   def this(format: Expression, timestamp: Expression) = this(format, timestamp, None)
 
   override def eval(input: InternalRow): Any = {
-    evalHelper(input, maxLevel = DateTimeUtils.TRUNC_TO_SECOND) { (t: Any, level: Int) =>
+    evalHelper(input, maxLevel = DateTimeUtils.TRUNC_TO_MICROSECOND) { (t: Any, level: Int) =>
```

Comment:
I'm maybe misreading this, but why is the 'max' now microseconds? does that disallow truncation to seconds?

Reply:
The `evalHelper()` and `codeGenHelper()` check that id of truncation level is in the valid range **[id <= 17 and id != -1]**: https://github.com/apache/spark/blob/caa23e3efd3f4422e8f599b30bec3ef1fb33c03c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L1406

They don't check limits semantically. We could assign special constants for the limit to not confuse readers in the future.

Reply:
Got it, I see you changed it, so I think the max is correct now